scripts/upgrade

```diff
@@ -16,45 +16,55 @@ ynh_script_progression --message="Loading installation settings..."
 
 app=$YNH_APP_INSTANCE_NAME
 
-server_domain=$(ynh_app_setting_get --app=$app --key=server_domain)
+domain=$(ynh_app_setting_get --app=$app --key=domain)
 port=$(ynh_app_setting_get --app=$app --key=port)
 game=$(ynh_app_setting_get --app=$app --key=game)
-is_public=$(ynh_app_setting_get --app=$app --key=is_public)
 final_path=$(ynh_app_setting_get --app=$app --key=final_path)
 home_path=$(ynh_app_setting_get --app=$app --key=home_path)
 pvp=$(ynh_app_setting_get --app=$app --key=pvp)
 creative=$(ynh_app_setting_get --app=$app --key=creative)
 damage=$(ynh_app_setting_get --app=$app --key=damage)
 servername=$(ynh_app_setting_get --app=$app --key=servername)
+is_public=$(ynh_app_setting_get --app=$app --key=is_public)
+
+#=================================================
+# CHECK VERSION
+#=================================================
+
+upgrade_type=$(ynh_check_app_version_changed)
 
 #=================================================
 # ENSURE DOWNWARD COMPATIBILITY
 #=================================================
 ynh_script_progression --message="Checking version..."
+
 version=ynh_app_upstream_version
 
 if [ "$version" = "0.01" ]; then
 ynh_die --message="You can't upgrade to this version with simple upgrade. Please read the readme for upgrading https://github.com/YunoHost-Apps/minetest_ynh#additional-information ..."
 fi
 
-
 ynh_script_progression --message="Ensuring downward compatibility..."
 
-# Fix is_public as a boolean value
-if [ "$is_public" = "Yes" ]; then
-	ynh_app_setting_set --app=$app --key=is_public --value=1
-	is_public=1
-elif [ "$is_public" = "No" ]; then
-	ynh_app_setting_set --app=$app --key=is_public --value=0
-	is_public=0
-fi
-
 # If final_path doesn't exist, create it
 if [ -z $final_path ]; then
 	final_path=/opt/yunohost/$app
 	ynh_app_setting_set --app=$app --key=final_path --value=$final_path
 fi
 
+# If domain doesn't exist, create it
+if [ -z "$domain" ]; then
+	domain=$server_domain
+	ynh_app_setting_set --app=$app --key=domain --value=$domain
+fi
```

Comment:
```suggestion
# If domain doesn't exist, create it
if [ -z "$domain" ]; then
	domain=$(ynh_app_setting_get --app=$app --key=server_domain)
	ynh_app_setting_set --app=$app --key=domain --value=$domain
fi
```

(Otherwise the var ain't defined ? :sweat_smile: )

Reply:
Yes, mmh, I was hoping for some magic to happen... 😅